### PR TITLE
Enable ENABLE_DEBUG option when CMAKE_BUILD_TYPE is Debug

### PR DESCRIPTION
**Pull request by matthiasklein**
This enables more verbose logging like:
STATE: DID => PERFORMING handle 0x56082ff9dd38; line 2218 (connection #0)

Signed-off-by: Matthias Klein <matthias@extraklein.de>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -222,6 +222,12 @@ else()
         include(cmake/zlib_external.cmake)
     endif()
 
+    string( TOLOWER "${CMAKE_BUILD_TYPE}" BUILD_TYPE_LOWER )
```
Comment (COM8) [inline CMakeLists.txt:225]:
```suggestion
    string(TOLOWER "${CMAKE_BUILD_TYPE}" BUILD_TYPE_LOWER)
```

**Review by COM8 — CHANGES_REQUESTED** [review]
Thanks for looking into this. Is this an additional level above [`CURLOPT_VERBOSE`](https://curl.se/libcurl/c/CURLOPT_VERBOSE.html) when it comes to debug output in libcurl?
Or is this flag used during build only?

@KingKili Should we promote this to a global cpr option since there are additional debug targets, like building with sanitizers and so on?

**Comment by matthiasklein** [comment]
> Or is this flag used during build only?
> 

Yes, the flag is used during Build only

**Comment by COM8** [comment]
Then in my eyes we should propagate this to a cpr option. What do you think about this @KingKili ?

**Comment by KingKili** [comment]
> Then in my eyes we should propagate this to a cpr option. What do you think about this @KingKili ?

I agree. Something like `curl_verbose_logging` makes more sense for me.

**Comment by COM8** [comment]
OK, then @matthiasklein could you please add a cpr option called `` wrapping ``
Example:
```cmake
```

**Comment by COM8** [comment]
OK, @matthiasklein could you please add a cpr option called something like `CURL_VERBOSE_LOGGING`, which wraps the curl `ENABLE_DEBUG` option?
```cmake
cpr_option(CURL_VERBOSE_LOGGING "Curl verbose logging during building curl ..." OFF)
```

**Comment by matthiasklein** [comment]
> OK, @matthiasklein could you please add a cpr option called something like `CURL_VERBOSE_LOGGING`, which wraps the curl `ENABLE_DEBUG` option?

Yes, I do

**Comment by matthiasklein** [comment]
Continued in the following PR: https://github.com/libcpr/cpr/pull/810

**Pull request closed without merging**